pkgs/servers/libreddit/default.nix

```
@@ -0,0 +1,34 @@
+{ lib
+, stdenv
+, nixosTests
+, rustPlatform
+, fetchFromGitHub
+, Security
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "libreddit";
+  version = "0.3.0";
```

libreddit at version 0.3.0 is running, but cannot fetch Reddit: `https://www.reddit.com/r/popular/hot.json?&raw_json=1 - Couldn't send request to Reddit: Head byte length should be less than 8kb`. The solution is to update the package.

```suggestion
  version = "0.10.1";
```

This version does no longer provide a `Cargo.lock`, so a patch seems required.

```diff
-  cargoSha256 = "0gk92bpqp462yrmnlkff311i6rgww4a9pan4xr435q2clajq79lw";
+  cargoSha256 = "0jx9sgdvp1x3hjrk1mqp52rfakmfgs5fgxz1hz4a3jj4nyrlx6k7";
+
+  cargoPatches = [
+    # a patch file to add/update Cargo.lock in the source code
+    ./add-Cargo.lock.patch
+  ];
```

With `add-Cargo.lock.patch` being <https://gist.github.com/erdnaxe/c4e25376e78f7996391d7eacccf939c1>.

I tested the module on NixOS and I am enthusiastic about this being merged in nixpkgs!

---

Set to 0.10.1 and added your changes. Thanks a lot.